src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2597,6 +2600,14 @@ def _get_website_node_version_functionapp(functions_version, runtime, runtime_ve
     return FUNCTIONS_VERSION_TO_DEFAULT_NODE_VERSION[functions_version]
 
 
+def _get_java_version_functionapp(functions_version, runtime_version):
+    if runtime_version is None:
+        runtime_version = FUNCTIONS_VERSION_TO_DEFAULT_RUNTIME_VERSION[functions_version]['java']
+    if runtime_version == '8':
```

Comment:
We need to default to Java 8 for  even if runtime version is not specified to ensure this is backward compatible.